libs/pavex/src/request/body/urlencoded.rs

```diff
@@ -0,0 +1,288 @@
+use crate::request::body::errors::{
+    ExtractUrlEncodedBodyError, MissingUrlEncodedContentType, UrlEncodedBodyDeserializationError,
+    UrlEncodedContentTypeMismatch,
+};
+use crate::request::body::BufferedBody;
+use crate::request::RequestHead;
+use http::HeaderMap;
+use serde::Deserialize;
+
+#[doc(alias = "UrlEncoded")]
+#[derive(Debug)]
+/// Parse the body of an incoming request as an application/x-www-form-urlencoded form.
```

Comment:
Most of this should go in a new guide section under [this one](https://pavex.dev/docs/guide/request_data/body/deserializers/)

Reply:
Just to clarify, shall I keep the comments in the code, and add a section under docs for urlencoded?

Reply:
I will look at how you have done JsonBody and mimic that one, to make it consistent.

Reply:
@LukeMathWalker how does this seem?